enclave_manager/avalon_enclave_manager/kme/enclave_manager.py

```diff
@@ -0,0 +1,201 @@
+        # model contracts expect byte32. Then take a hexdigest for hex str.
+        worker_id = hashlib.sha256(worker_id).hexdigest()
+        self._worker_kv_delegate.add_new_worker(worker_id, worker_info)
+
+        # Cleanup wo-processing" table
+        self._wo_kv_delegate.cleanup_work_orders()
+
+# -------------------------------------------------------------------------
+
+    def _execute_work_order(self, input_json_str, indent=4):
+        """
+        Submits request to KME and retrieves the response
+
+        Parameters :
+            input_json_str - A JSON formatted str of the request to execute
+        Returns :
+            json_response - A JSON formatted str of the response received from
+                            the enclave
+        """
+        try:
+            wo_request = work_order_request.SgxWorkOrderRequest(
+                self.enclave_data,
+                input_json_str)
+            wo_response = wo_request.execute()
+
+            try:
+                json_response = json.dumps(wo_response, indent=indent)
+            except Exception as err:
+                logger.error("ERROR: Failed to serialize JSON; %s", str(err))
+
+        except Exception as e:
+            logger.error("failed to execute work order; %s", str(e))
+
+        return json_response
+
+# -------------------------------------------------------------------------
+
+    def start_enclave_manager(self):
+        """
+        Execute boot flow and run time flow
+        """
+        try:
+            logger.info(
+                "--------------- Starting Boot time flow ----------------")
+            self._manager_on_boot()
+            logger.info(
+                "--------------- Boot time flow Complete ----------------")
+        except Exception as err:
+            logger.error("Failed to execute boot time flow; " +
+                         "exiting Intel SGX Enclave manager: {}".format(err))
+            exit(1)
+
+        self._start_kme_listener()
+
+
+# -------------------------------------------------------------------------
+
+    def _start_kme_listener(self):
+        """
+        This function reads the host_name and port for the listener to
+        bind. It the instantiates the KMEListener and starts it. This
+        listener runs indefinitely processing requests received from
+        WPEs. It terminates only when an exception occurs within start
+        method os the listener instance.
+        """
+        host_name, port = parse_bind_url(
+            self._config["KMEListener"].get("bind"))
+
+        rpc_methods = [
+            self.GetUniqueVerificationKey,
+            self.RegisterWorkorderProcessor,
+            self.PreporocessWorkorder
+        ]
+        kme_listener = KMEListener(rpc_methods)
+        kme_listener.start(host_name, port)
+
+
+# -----------------------------------------------------------------
+
+    def GetUniqueVerificationKey(self, **parms):
```

Comment:
Is this function exposed outside? If its meant to be used internally, you can prepend function with underscore. Same thing applies to other methods as well.

Reply:
Renamed as of now though I guess I might want to move it to the listener. Sounds more logical to have the handlers along side the listener.